src/Classes/TradeQueryGenerator.lua

```diff
@@ -0,0 +1,695 @@
+            local modValueStr = (entry.sign and entry.sign or "") .. tostring(modValue)
+
+            -- Apply override text for special cases
+            local modLine
+            if modValue == 1 and entry.specialCaseData.overrideModLineSingular ~= nil then
+                modLine = entry.specialCaseData.overrideModLineSingular
+            elseif entry.specialCaseData.overrideModLine ~= nil then
+                modLine = entry.specialCaseData.overrideModLine
+            else
+                modLine = entry.tradeMod.text
+            end
+            modLine = modLine:gsub("#",modValueStr)
+
+            self.calcContext.testItem.explicitModLines[1] = { line = modLine, custom = true }
+            self.calcContext.testItem:BuildAndParseRaw()
+
+            if (self.calcContext.testItem.modList ~= nil and #self.calcContext.testItem.modList == 0) or (self.calcContext.testItem.slotModList ~= nil and #self.calcContext.testItem.slotModList[1] == 0 and #self.calcContext.testItem.slotModList[2] == 0) then
+                logToFile("Failed to test %s mod: %s", self.calcContext.itemCategory, modLine)
+            end
+
+            local output = self.calcContext.calcFunc({ repSlotName = self.calcContext.slot.slotName, repItem = self.calcContext.testItem }, {})
+            local meanDPSDiff = (GlobalCache.useFullDPS and output.FullDPS or output.TotalDPS) - self.calcContext.baseDPS
+            if meanDPSDiff > 0.01 then
+                table.insert(self.modWeights, { tradeModId = entry.tradeMod.id, weight = meanDPSDiff / modValue, meanDPSDiff = meanDPSDiff, invert = entry.sign == "-" and true or false })
+                self.alreadyWeightedMods[entry.tradeMod.id] = true
+            end
+
+            local now = GetTime()
+			if now - start > 50 then
+                -- Would be nice to update x/y progress on the popup here, but getting y ahead of time has a cost, and the visual seems to update on a significant delay anyways so it's not very useful
+				coroutine.yield()
+				start = now
+			end
+        end
+        ::continue::
+    end
+end
+
+function TradeQueryGeneratorClass:OnFrame()
+    if self.calcContext.co == nil then
+        return
+    end
+
+    local res, errMsg = coroutine.resume(self.calcContext.co, self)
+    if launch.devMode and not res then
+        error(errMsg)
+    end
+    if coroutine.status(self.calcContext.co) == "dead" then
+        self.calcContext.co = nil
+        self:FinishQuery()
+    end
+end
+
+function TradeQueryGeneratorClass:StartQuery(slot, options)
+    -- Figure out what type of item we're searching for
+    local existingItem = self.itemsTab.items[slot.selItemId]
+    local testItemType = existingItem and existingItem.baseName or "Unset Amulet"
+    local itemCategoryQueryStr
+    local itemCategory
+    if slot.slotName == "Weapon 2" or slot.slotName == "Weapon 1" then
+        if existingItem.type == "Shield" then
+            itemCategoryQueryStr = "armour.shield"
```

Comment:
This string of ifs might be more appropriate as a table.  Ideally generated, but I'm not sure if that is easy to do.